PYIC-7129: Remove old cimit lambda code

Wynndow opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

Remove old cimit lambda code

### Why did it change

We no longer call the cimit lambdas directly, instead using an API gateway. This means we can remove all the cruft.

### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYIC-7129](https://govukverify.atlassian.net/browse/PYIC-7129)


[PYIC-7129]: https://govukverify.atlassian.net/browse/PYIC-7129?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Wynndow committed:
PYIC-7129: Remove old cimit lambda code

We no longer call the cimit lambdas directly, instead using an API
gateway. This means we can remove all the cruft.

Wynndow committed:
BAU: Fix template violations

GHA is using a new vesrion of sam, which picks up more tempalte
violations.

Mostly around the use of if and having dead code paths.
The change to the alarm resource is because you shouldn't define
dimensions if you're defining metrics.

Joe-Edwards-GDS commented on libs/cimit-service/src/main/java/uk/gov/di/ipv/core/library/service/CiMitService.java:
```diff
@@ -437,9 +284,13 @@ private HttpResponse<String> sendHttpRequest(HttpRequest cimitHttpRequest)
         } catch (InterruptedException e) {
+            LOGGER.error(
+                    LogHelper.buildErrorMessage(
+                            "Client interrupted sending http request to CiMit", e));
```
I know we have it inconsistent throughout our code, but can we try to align on `CIMIT` as a stylisation going forward? (Likewise `SPOT` over `SPoT`)

I asked Chris C and he agrees - he thinks `CiMit` might have been a Slade-ism.

Wynndow replied:
ha - yep. Good to have an offical steer.

Joe-Edwards-GDS commented on libs/cimit-service/src/main/java/uk/gov/di/ipv/core/library/service/CiMitService.java:
```diff
@@ -128,34 +86,9 @@ public void submitVC(VerifiableCredential vc, String govukSigninJourneyId, Strin
+
+            sendPostHttpRequest(POST_CI_ENDPOINT, govukSigninJourneyId, ipAddress, payload);
         } catch (JsonProcessingException e) {
             throw new CiPutException("Failed to serialize payload for post CI request.");
```
One thing that Patrick spotted (https://gds.slack.com/archives/C05TJ52FZ8B/p1724411977038429) is that this error can also be thrown when deserializing the response and the error message is quite misleading.

The other thing from both these blocks is that we're losing the underlying exception, so we should make sure any relevant detail has been logged.

Joe-Edwards-GDS reviewed (commented):
1000 lines deleted 😍

Wynndow committed:
PYIC-7129: Remove use of CiMit

Instead use cimit, Cimit or CIMIT, depending.

Wynndow committed:
PYIC-7129: Ensure error details are logged

Adds logging to ensure that we're not losing detais when catching errors
and rethrowing.
